### PR TITLE
adds a cybernetic eyepatch to the syndicate uplink

### PR DESCRIPTION
**Pull request by timothyteakettle**
## About The Pull Request
this adds a mechanical eyepatch to the uplink for 8 tc
it makes any bullet you shoot have incredibly high ricochet stats so they'll bounce off walls and into people
(also fixes ricochets)

the full stats are as follows:
gives bullets 10 ricochet max, 100% ricochet chance with 0 ricochet chance decay, 360 degree aim seeking but only if a target is within 3 tiles, also bullets lose 10% of their damage after every ricochet

i threw the ricochet fix ontop because it was needed to make it work but if it isn't accepted i'll make a separate pr that fixes ricochets

also it removes the need to be in combat mode to have normal accuracy while worn, and there's a little special thing for people who have poor aim and use it that makes stuff interesting

## Why It's Good For The Game
haha funny eyepatch make bullets go brrr

## Changelog
:cl:
add: syndicate agents now have access to mechanical aim enhancers which allow them to aim bullets to bounce off walls
fix: ricochets work properly now for the bullets that support them
/:cl:

### CODE REVIEW

**Comment by Arturlang** [comment]
Neednt be so expensive for how situational it is

**Comment by timothyteakettle** [comment]
it's **very** powerful but I am unsure how situational ricochets are so maybe

**Comment by timothyteakettle** [comment]
lowered cost to 8 tc

File: code/modules/clothing/glasses/_glasses.dm [hunk 1]
```diff
@@ -149,6 +149,29 @@
 	icon_state = "eyepatch"
 	item_state = "eyepatch"
 
+/obj/item/clothing/glasses/eyepatch/syndicate
+	name = "cybernetic eyepatch"
+	desc = "An eyepatch used to enhance one's aim with guns."
+	icon_state = "syndicatepatch"
+	item_state = "syndicatepatch"
+
```
Comment (silicons) [inline code/modules/clothing/glasses/_glasses.dm:158]: acidproof it or something idk

**Comment by lolman360** [comment]
7 so you can use it with the revolver please

**Comment by timothyteakettle** [comment]
purposefully 8 to stop it being a meta go-to for traitors who want to game end security
also hey don't merge this yet kevin wants me to not scrap the ricochet stuff i removed so he's apparently going to look at why it wasn't working (?)

**Comment by timothyteakettle** [comment]
okay the changes are done it's cool now even though the old stuff doesn't really make sense for why it exists but whatever
edit: changed it again to fit better everythings cool now and bullets ricochet off way more things

**Review by silicons — COMMENTED** [review]
changing how ricochet flags are done isn't necessary, just make the eyepatch give the projectile a flag to make it always ricochet (or variable if you want for number of forced ricochets)

i don't entirely like how i handled ricochets but the entire system has a way to override whether or not projectiles ricochet from the projectile since projectile gets final say. you just need to tell projectile check_ricochet to return TRUE as long as a forced ricochet is left, and for on_ricochet or whatever to minus 1 ricochet from forced ricochets if it's above 0 to make it not ricochet forever.

**Comment by silicons** [comment]
alright fuck it i'll code jannie this later :/

**Pull request merged**